ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/HikariJdbcConnectionProvider.scala

```diff
@@ -43,9 +45,12 @@ object HikariConnection {
       config.setPoolName(s"$ConnectionPoolPrefix.${serverRole.threadPoolSuffix}")
       metrics.foreach(config.setMetricRegistry)
 
-      //note that Hikari uses auto-commit by default.
-      //in `runSql` below, the `.close()` will automatically trigger a commit.
-      new HikariDataSource(config)
+      // Hikari dies if a database connection could not be opened almost immediately
+      // regardless of any connection timeout settings. We retry connections so that
+      // Postgres and Sandbox can be started in any order.
+      RetryStrategy.constant(attempts = 600, waitTime = 1.second) { (_, _) =>
+        Future { new HikariDataSource(config) }
```

Comment:
So I don't have an `ExecutionContext` here. I _think_ the appropriate one is the one passed in with the call to `ResourceOwner.acquire`, but this `ExecutionContext` is not exposed here.

Would you prefer I:
* Modify `FutureCloseableResourceOwner` to expose the `ExecutionContext` to this function body? It involves changing the other usages to ignore this parameter now too.
* Implement an explicit `ResourceOwner` subclass whose `acquire` method is essentially the body of this method? This approach seems to make the most sense to me, though it does bloat the code a little here.
* Create a single-threaded `ExecutionContext` _just_ for this retry here?
* Add `ExecutionContext` as an implicit parameter to the `owner` method? (This seems completely wrong to me if I'm understanding what `ResourceOwner` is meant to accomplish though).

Reply:
(I guess scalaz sort of hid that decision from us, which is definitely a strong argument for not using it here)

Reply:
I think you want to add an explicit `ResourceOwner`. I'm increasingly thinking that some of those helper constructors were a mistake.

Reply:
It's important that the execution context is passed in during acquisition, not when constructing the owner. I tried the latter and it caused a lot of subtle problems.

Reply:
@SamirTalwar got around to finishing this up and made the changes as discussed.

Reply:
Awesome.